File: Core/src/main/java/jokatu/game/stage/SingleInputStage.java
package jokatu.game.stage;

import jokatu.game.event.GameEvent;
import jokatu.game.input.AbstractInputAcceptor;
import jokatu.game.input.Input;
import jokatu.game.player.Player;

/**
 * A {@link Stage} that only accepts one kind of {@link Input} from one kind of {@link Player}.
 */
public abstract class SingleInputStage<I extends Input, P extends Player, E extends GameEvent>
		extends AbstractInputAcceptor<I, P, E> implements Stage<E> {
}

File: Core/src/main/kotlin/jokatu/game/stage/SingleInputStage.kt
package jokatu.game.stage

import jokatu.game.event.GameEvent
import jokatu.game.input.AbstractInputAcceptor
import jokatu.game.input.Input
import jokatu.game.player.Player
import kotlin.reflect.KClass

/**
 * A [Stage] that only accepts one kind of [Input] from one kind of [Player].
 */
abstract class SingleInputStage<I : Input, P : Player, E : GameEvent>(
		inputClass: KClass<I>,
		playerClass: KClass<P>
) : AbstractInputAcceptor<I, P, E>(inputClass, playerClass), Stage<E>
